tests/cancun/eip1153_tstore/test_basic_tload.py

```diff
@@ -0,0 +1,345 @@
+
+    address_to = Address("A00000000000000000000000000000000000000A")
+    pre = {
+        address_to: Account(
+            balance=1000000000000000000,
+            nonce=0,
+            code=Op.JUMPDEST()
+            # 03 test
+            + Op.TSTORE(3, tstore_value)
+            + Op.SSTORE(slot_tload_untouched_slot_after_tstore_result, Op.TLOAD(0))
+            + Op.SSTORE(slot_code_worked, 1),
+            storage={
+                slot_tload_untouched_slot_after_tstore_result: 0xFF,
+            },
+        ),
+        TestAddress: Account(
+            balance=7000000000000000000,
+            nonce=0,
+            code="0x",
+            storage={},
+        ),
+    }
+
+    post: Dict[Address, Union[Account, object]] = {}
+
+    post[address_to] = Account(
+        storage={
+            slot_tload_untouched_slot_after_tstore_result: 0x00,
+            slot_code_worked: 0x01,
+        }
+    )
+
+    tx = Transaction(
+        nonce=0,
+        to=address_to,
+        gas_price=10,
+        data=b"",
+        gas_limit=5000000,
+        value=0,
+    )
+
+    state_test(env=Environment(), pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Cancun")
+def test_basic_tload_gasprice(
+    state_test: StateTestFiller,
+):
+    """
+    Ported .json vectors:
+
+    (16_tloadGasFiller.yml)
+    tload costs 100 gas same as a warm sload
+    """
+    slot_tload_nonzero_gas_price_result = 1
+    slot_tload_zero_gas_price_result = 2
+    slot_code_worked = 3
+
+    address_to = Address("A00000000000000000000000000000000000000A")
+
+    """
+    N         OPNAME       GAS_COST  TOTAL_GAS REMAINING_GAS     STACK
+    28-1         MSTORE         2     20748   4958252    2:[4ba82f,0,]
+                 MSTORE [0] = 4958255
+    29-1          PUSH1         3     20754   4958246
+    30-1          TLOAD       100     20757   4958243    1:[10,]
+    31-1            GAS         2     20857   4958143    1:[2,]
+    32-1          PUSH1         3     20859   4958141    2:[2,4ba7bd,]
+    33-1         MSTORE         6     20862   4958138    3:[2,4ba7bd,20,]
+                 MSTORE [32] = 4958141
+    """
+    extra_opcode_gas = 11  # mstore(3), push1(3),gas(2),push1(3)
+
+    pre = {
+        address_to: Account(
+            balance=1000000000000000000,
+            nonce=0,
+            code=Op.JUMPDEST()
+            # 16 test
+            + Op.TSTORE(16, 2)
+            + Op.MSTORE(0, Op.GAS())  # hot load the memory to make the extra_opcode_gas be 11
+            + Op.MSTORE(0, Op.GAS())
+            + Op.TLOAD(16)
+            + Op.MSTORE(32, Op.GAS())
+            + Op.SSTORE(slot_tload_nonzero_gas_price_result, Op.SUB(Op.MLOAD(0), Op.MLOAD(32)))
+            + Op.SSTORE(
+                slot_tload_nonzero_gas_price_result,
+                Op.SUB(Op.SLOAD(slot_tload_nonzero_gas_price_result), extra_opcode_gas),
+            )
+            # from zero slot
+            + Op.MSTORE(0, Op.GAS())
+            + Op.TLOAD(5)
```

Comment:
Does the comment above mean that we should be testing `Op.TLOAD(0)` here? It's a bit confusing otherwise.

Reply:
no mean tload from slot that was set to 0